source/functions/handle-errors.txt

```diff
@@ -0,0 +1,475 @@
+          - Rest parameters
+          - Indefinite number of arguments passed to the main function.
+
+      ``handleRetry`` performs the following operations:
+
+      #. Attempts to execute ``functionToRetry`` in a ``try`` statement.
+         If the execution is successful, ``handleRetry`` returns the value
+         returned by ``functionToRetry``.
+      #. If the execution of ``functionToRetry`` in the previous step throws an error,
+         the ``catch`` statement handles the error as follows:
+
+         a. Checks if the number of previous retries equals the
+            maximum permitted number of retries.
+            If the two numbers are the same, then the function throws an error
+            because the max retries has been reached. The function no longer attempts
+            to retry.
+         #. Build a function execution log entry object to insert into the database.
+         #. Get a reference to the failed execution tracker collection.
+         #. Insert the function log exection log entry into
+            the failed execution tracker collection. This insertion operation
+            causes the Database Trigger Function, which you will make in the next step,
+            to fire.
+
+      The main function is passed as the argument ``functionToRetry``.
+      ``handleRetry`` attempts to execute the main Function.
+      If the execution fails, this function attempts to retry the main function.
+
+      .. tabs-realm-admin-interfaces::
+
+         .. tab::
+            :tabid: ui
+
+            Navigate to :guilabel:`Functions`. Click the button :guilabel:`Create New Function`.
+
+            In the field :guilabel:`Name`, add ``handleRetry``.
+
+            In the :guilabel:`Function Editor` add the following code,
+            then save the Function:
+
+            .. literalinclude:: /examples/generated/functions/handleRetry.snippet.handle-retry.js
+               :language: js
+               :caption: handleRetry.js
+
+         .. tab::
+            :tabid: cli
+
+            .. tip:: Install and Set up Realm CLI
+
+               If you're using the Realm CLI to update your App Services App,
+               you must first :ref:`install and set up the Realm CLI <realm-cli>`.
+
+            Add the following to :file:`functions/config.json`:
+
+            .. code-block:: js
+               :caption: functions/config.json
+
+               [
+                 {
+                   "name": "handleRetry",
+                   "private": true,
+                   "run_as_system": true
+                 }
+                 // ...other configuration
+               ]
+
+            Create the file for the Function :file:`functions/handleRetry.js`:
+
+            .. literalinclude:: /examples/generated/functions/handleRetry.snippet.handle-retry.js
+               :language: js
+               :caption: functions/handleRetry.js
+
+            Push your changes to App Services:
+
+            .. code-block: sh
```

Comment:
This code block isn't showing up on the page, and I _think_ this is the reason why?

```suggestion
            .. code-block:: sh
```